### PR TITLE
Slideshow Block: UI Improvements

### LINKED ISSUE #30672: Slideshow: Outstanding issues
This issue is to keep track of and discuss the outstanding issues with the slideshow block.

1) The appearance of the left/right arrows. 
The main issue here is that on a while background they are impossible to see. To address this we should add a strong dark shadow beneath them.

2) The placement and appearance of the pagination dots.
I think these are fine as they are. The only change I'd make is to make the containing link a bit larger, so that I can click on them easier.

3) Placement and appearance of captions.
We could put the captions above the pagination dots. White text on a dark translucent background. Centrally aligned. The text would need to get smaller as the block shrunk.

4) How to handle captions that exceed 1 line in width.
I think they can just keep growing

5) The resizing "algorithm"
I think we can do a bit more work here. I added some images which all had the same aspect ratio, but there was still a bit of background showing through. In cases where all the images are the same size we should make the slideshow the same aspect ratio as the images. In cases where they aren't, can we crop the images to fit?

6) Background color behind images that don't fill the whole area.

Ideally the images would fill the whole area. Where this is not possible, can we made the background transparent, so that the theme's background shows through?

Keyboard navigation: Users should be able to navigate the slideshow using their keyboard. Left and right buttons make the most sense. (see https://www.w3.org/WAI/tutorials/carousels/)

Let me know if I can help with any of these changes.

cc @davidakennedy for an accessibility perspective
cc @thomasguillot as he always has something useful to say :)

### PR DESCRIPTION
**Pull request by jeffersonrabb**
#### Changes proposed in this Pull Request

This PR contains fixes to design/UI issues raised in https://github.com/Automattic/wp-calypso/issues/30672. 

1. Arrows
> I would suggest to add a background-color to `.wp-block-jetpack-slideshow .wp-block-jetpack-slideshow_button-prev, .wp-block-jetpack-slideshow .wp-block-jetpack-slideshow_button-next`. Something like `rgba(0, 0, 0, 0.5)`.

Done in https://github.com/Automattic/wp-calypso/pull/30687/commits/3487da374cbc118c24599de970c00073107b066d

2. Pagination Style and Placement
> I'd be tempted to have them outside of the actual slider. I would also increase the size and tweak the style a bit. I think it'd be cool if we could customise the colour of the active bullet.

Done in https://github.com/Automattic/wp-calypso/pull/30687/commits/b65b6f54b100a45828dd67e25c045172ff4ccaba

3. Captions
> Move the caption into the actual slider. Similar to the arrows, white text with a dark background `rgba(0, 0, 0, 0.5)`.

Done in https://github.com/Automattic/wp-calypso/pull/30687/commits/b65b6f54b100a45828dd67e25c045172ff4ccaba

4. Keyboard Navigation
> Keyboard navigation: Users should be able to navigate the slideshow using their keyboard. Left and right buttons make the most sense. (see https://www.w3.org/WAI/tutorials/carousels/)

Done (for front end only) in https://github.com/Automattic/wp-calypso/pull/30687/commits/c2a82ee6a2e8e0992e62d84897ce21baa25ed6ba

#### Testing instructions

1. Spin up Jurassic Ninja instance: https://jurassic.ninja/create/?gutenpack&calypsobranch=update/slider-ui-changes
2. Connect Jetpack
3. Enable `JETPACK_BETA_BLOCKS` in `Settings->Jetpack Constants`.
4. Create Post, add Slideshow, upload images.
5. Verify UI changes listed above.

Fixes #30672

### CODE REVIEW

**Comment by thomasguillot** [comment]
There's a bug with the caption with the "Flip" transition: https://cloudup.com/cPkCtek0Hwd

Maybe we can just remove this transition?

**Comment by thomasguillot** [comment]
One feature I'm missing is the ability to loop the slides, could be a new setting?

**Comment by jeffersonrabb** [comment]
> One feature I'm missing is the ability to loop the slides, could be a new setting?

You should see an `Autoplay` toggle in the block sidebar which, when enabled, reveals a range control:

screen shot 2019-02-11 at 8 53 04 am

**Comment by jeffersonrabb** [comment]
> Maybe we can just remove this transition?

Sure. While I'm at it, @thomasguillot what do you think of the Cover Flow transition? In the original Atavist block, we only offered Slide and Fade, because we found the others to be a bit 🧀-y. I included them all in the initial version of the Gutenberg block just so we could assess.

**Comment by jeffersonrabb** [comment]
@thomasguillot, for https://github.com/Automattic/wp-calypso/pull/30687#commitcomment-32266252 and https://github.com/Automattic/wp-calypso/pull/30687#commitcomment-32266294, which element are you referring to?

**Comment by thomasguillot** [comment]
> You should see an `Autoplay` toggle in the block sidebar which, when enabled, reveals a range control.

I meant when I'm at the end of my slideshow, If I want to keep clicking on the right arrow and loop back to the beginning. 

screenshot 2019-02-11 at 13 57 35

**Comment by thomasguillot** [comment]
`.wp-block-jetpack-slideshow_caption` for https://github.com/Automattic/wp-calypso/pull/30687#commitcomment-32266252

`.swiper-pagination-bullet-active` for https://github.com/Automattic/wp-calypso/pull/30687#commitcomment-32266294

**Comment by thomasguillot** [comment]
> Sure. While I'm at it, @thomasguillot what do you think of the Cover Flow transition? In the original Atavist block, we only offered Slide and Fade, because we found the others to be a bit 🧀-y. I included them all in the initial version of the Gutenberg block just so we could assess.

Haha yeah I agree. I'm ok removing it but I wonder what @scruffian thinks about this.

**Comment by jeffersonrabb** [comment]
> I meant when I'm at the end of my slideshow, If I want to keep clicking on the right arrow and loop back to the beginning.

Hmmm, I don't see anything in Swiper's API about this, which is a little surprising. As a gut check, do you see anything about this I'm missing? https://idangero.us/swiper/api/

**Comment by thomasguillot** [comment]
> > I meant when I'm at the end of my slideshow, If I want to keep clicking on the right arrow and loop back to the beginning.
> 
> Hmmm, I don't see anything in Swiper's API about this, which is a little surprising. As a gut check, do you see anything about this I'm missing? https://idangero.us/swiper/api/

In [Methods & Properties](https://idangero.us/swiper/api/#methods), I see a loop parameter, could it be it?

screenshot 2019-02-11 at 14 46 10

**Comment by scruffian** [comment]
> Haha yeah I agree. I'm ok removing it but I wonder what @scruffian thinks about this.

I say lets remove them and see if anyone asks for them...

**Comment by jeffersonrabb** [comment]
> I say lets remove them and see if anyone asks for them...

Removed in https://github.com/Automattic/wp-calypso/pull/30687/commits/ee4543b85e2541687fb402234e627099211db5c4

**Comment by scruffian** [comment]
I added 2762455 to update the captions and bullets:

screenshot 2019-02-11 at 17 48 29

**Comment by jeffersonrabb** [comment]
> In [Methods & Properties](https://idangero.us/swiper/api/#methods), I see a loop parameter, could it be it?

Yup, that was it. It's added in https://github.com/Automattic/wp-calypso/pull/30687/commits/18bc08959fd72da7708ccd0cfffe4426f2acf0a2.

File: client/gutenberg/extensions/slideshow/style.scss [hunk 1]
```diff
@@ -8,21 +8,19 @@
...
 	}
 
 	.wp-block-jetpack-slideshow_slide {
-		margin: 0;
-	}
-
-	.wp-block-jetpack-slideshow_image-container {
 		align-items: center;
-		background: #f0f0f0;
+		background: rgba( 0, 0, 0, 0.1 );
```
Comment (scruffian) [inline client/gutenberg/extensions/slideshow/style.scss]: Why not just make this `background: transparent`?

Comment (thomasguillot) [reply client/gutenberg/extensions/slideshow/style.scss]: To have a very subtle background that determines the boundaries of the slideshow and make it a complete block. A transparent background would make the arrows almost "floating" if the image wasn't wide enough.

**Comment by jeffersonrabb** [comment]
> even gets us rid of that extra `<div />`

Note: I closed https://github.com/Automattic/wp-calypso/pull/30629 as a result.

**Pull request merged**